### DISCUSSION TITLE
assembler output addition

### DISCUSSION THREAD

**Discussion by udo-munk** [udo-munk/z80pack#271] [Ideas]
It would be helpfull if the assembler directly could output a C array with the code, that can be used for IoT device memory, without the need to convert first from a binary or hex.

**Comment by sneakywumpus** [comment]
Implemented in #272

**Comment by udo-munk** [comment]
Very fast, thanks!

But code cannot be const, a von Neumann CPU can also write into code memory:

C:\Users\udomu\Documents\Arduino\arduino_8080\memory.h: In function 'void memwrt(WORD, BYTE)':
C:\Users\udomu\Documents\Arduino\arduino_8080\memory.h:26:18: error: assignment of read-only location 'code[addr]'
     code[addr] = data;
                  ^~~~

exit status 1

Compilation error: assignment of read-only location 'code[addr]'

**Comment by udo-munk** [comment]
Works now, very nice and easy to produce code now for the Arduino8080 :-)

**Comment by udo-munk** [comment]
There is another funny warning from GCC, this is Ubuntu Linux on Windows 11 WSL

udo@nb-udo:~/work/z80pack/z80asm$ make
cc -O3 -Wall -Wextra -Wwrite-strings -D_POSIX_C_SOURCE=200809L -c z80amain.c
z80amain.c: In function ‘options’:
z80amain.c:246:34: warning: argument 1 range [18446744056529682432, 18446744073709551608] exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
  246 |         if ((infiles = (char **) malloc(sizeof(char *) * nfiles)) == NULL)
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from z80amain.c:11:
/usr/include/stdlib.h:540:14: note: in a call to allocation function ‘malloc’ declared here
  540 | extern void *malloc (size_t __size) __THROW __attribute_malloc__
      |              ^~~~~~

**Reply by udo-munk** [reply]
Fixed in d8ae803

**Comment by udo-munk** [comment]
Changed the produced C code a bit, I do not need a code[] array with the code only, but a memory immage. Besides the code I need a bit RAM for the stack, data and so on. While it seemed to work on AVR platform the behaviour was a bit undefined and risky ;-)
Also I like your idea to provide informations about code[] in variables very much. But for now not needed and every single byte consumed memory counts. So I made the variables comments, then one can leave as is and save the bytes, or uncomment if really needed.
